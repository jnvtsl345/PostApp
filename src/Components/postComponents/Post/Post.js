import React from 'react';
import postStyle from './Post.module.css';
import PropTypes from 'prop-types';
function Post({ title, content, author }){
    return(
        <div className={postStyle.post}>
            <h3 data-testid="post-title">{title}</h3>
            <p data-testid="post-content">{content}</p>
            <label data-testid="post-author">Created By:<strong> {author}</strong></label>
        </div>
    )
}

Post.propTypes={
    title:PropTypes.string,
    content:PropTypes.string,
    author:PropTypes.string
}
export default Post;