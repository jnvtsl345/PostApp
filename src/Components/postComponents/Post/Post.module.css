.post{
    padding-bottom:27px;
}
.post label{
    float:right;
    margin-right: 30px;
}