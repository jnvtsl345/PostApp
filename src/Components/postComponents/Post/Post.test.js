import React from 'react';
import ReactDom from 'react-dom';
import Post from './Post.js';

import { render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import "@testing-library/jest-dom/extend-expect";


afterEach(cleanup);

it("renders component without crashing", () =>{
    const div =document.createElement("div");
    ReactDom.render(<Post ></Post>,div);
})

it("renders post correctly", () =>{
    const post ={ title:"Soccer" ,content:"Best game ever!!", author:"Vatsal"};
    const {getByTestId} = render(<Post {...post} ></Post>)
    expect(getByTestId('post-title')).toHaveTextContent("Soccer")
    expect(getByTestId('post-content')).toHaveTextContent("Best game ever!!")
    expect(getByTestId('post-author')).toHaveTextContent("Vatsal")
})

it("matches snapshot",()=>{
    const post ={ title:"Football" ,content:"Best game ever!!", author:"Vatsal"};
    const tree = renderer.create(<Post {...post}></Post>).toJSON();
    expect(tree).toMatchSnapshot();
})