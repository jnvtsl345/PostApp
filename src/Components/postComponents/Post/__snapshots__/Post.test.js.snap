// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="post"
>
  <h3
    data-testid="post-title"
  >
    Football
  </h3>
  <p
    data-testid="post-content"
  >
    Best game ever!!
  </p>
  <label
    data-testid="post-author"
  >
    Created By:
    <strong>
       
      Vatsal
    </strong>
  </label>
</div>
`;
